testing/trino-product-tests/src/main/java/io/trino/tests/product/jdbc/TestKerberosConstrainedDelegationJdbc.java

```diff
@@ -0,0 +1,168 @@
+        extends ProductTest
+{
+    private static final String KERBEROS_OID = "1.2.840.113554.1.2.2";
+    @Inject
+    @Named("databases.presto.jdbc_url")
+    String jdbcUrl;
+
+    @Inject
+    @Named("databases.presto.kerberos_principal")
+    private String kerberosPrincipal;
+
+    @Inject
+    @Named("databases.presto.kerberos_keytab")
+    private String kerberosKeytab;
+
+    private GSSManager gssManager;
+    private KerberosAuthentication kerberosAuthentication;
+
+    @BeforeMethodWithContext
+    public void setUp()
+    {
+        this.gssManager = GSSManager.getInstance();
+        this.kerberosAuthentication = new KerberosAuthentication(kerberosPrincipal, kerberosKeytab);
+    }
+
+    @Test(groups = JDBC_KERBEROS_CONSTRAINED_DELEGATION)
+    public void testSelectConstrainedDelegationKerberos()
+            throws Exception
+    {
+        Properties driverProperties = new Properties();
+        GSSCredential credential = createGssCredential();
+        driverProperties.put("KerberosConstrainedDelegation", credential);
+        try (Connection connection = DriverManager.getConnection(jdbcUrl, driverProperties);
+                PreparedStatement statement = connection.prepareStatement("SELECT * FROM tpch.tiny.nation");
+                ResultSet results = statement.executeQuery()) {
+            assertThat(forResultSet(results)).matches(TpchTableResults.PRESTO_NATION_RESULT);
+        }
+        finally {
+            credential.dispose();
+        }
+    }
+
+    @Test(groups = JDBC_KERBEROS_CONSTRAINED_DELEGATION)
+    public void testCtasConstrainedDelegationKerberos()
+            throws Exception
+    {
+        Properties driverProperties = new Properties();
+        GSSCredential credential = createGssCredential();
+        driverProperties.put("KerberosConstrainedDelegation", credential);
+        try (Connection connection = DriverManager.getConnection(jdbcUrl, driverProperties);
+                PreparedStatement statement = connection.prepareStatement(format("CREATE TABLE %s AS SELECT * FROM tpch.tiny.nation", "test_kerberos_ctas"))) {
+            int results = statement.executeUpdate();
+            Assertions.assertThat(results).isEqualTo(25);
+        }
+        finally {
+            credential.dispose();
+        }
+    }
+
+    @Test(groups = JDBC_KERBEROS_CONSTRAINED_DELEGATION)
+    public void testSelectConstrainedDelegationKerberosDisposed()
+            throws Exception
+    {
+        Properties driverProperties = new Properties();
+        GSSCredential credential = createGssCredential();
+        credential.dispose();
+        driverProperties.put("KerberosConstrainedDelegation", credential);
+        try (Connection connection = DriverManager.getConnection(jdbcUrl, driverProperties);
+                PreparedStatement statement = connection.prepareStatement("SELECT * FROM tpch.tiny.nation");
+                ResultSet results = statement.executeQuery()) {
+            forResultSet(results);
```

Comment:
Can we use `assertThatThrownBy` to check for failure ?

Reply:
refactored